server/methods/core/cart.js

```diff
@@ -434,16 +434,45 @@ Meteor.methods({
     // we need to get the parent of the option to check if parcel info is stored there
     const immediateAncestors = variant.ancestors.filter((ancestor) => ancestor !== product._id);
     const immediateAncestor = Collections.Products.findOne({ _id: immediateAncestors[0] });
-    let parcel = null;
-    if (immediateAncestor) {
-      if (immediateAncestor.weight || immediateAncestor.height || immediateAncestor.width || immediateAncestor.length) {
-        parcel = { weight: immediateAncestor.weight, height: immediateAncestor.height, width: immediateAncestor.width, length: immediateAncestor.length };
-      }
+
+    // Get maketplace settings
+    const marketplaceSettings = Reaction.getMarketplaceSettings();
+
+    let shopId;
+    if (marketplaceSettings && marketplaceSettings.public && marketplaceSettings.public.merchantCart) {
+      shopId = Reaction.getShopId();
+    } else {
+      shopId = Reaction.getPrimaryShopId();
     }
-    // if it's set at the option level then that overrides
-    if (variant.weight || variant.height || variant.width || variant.length) {
-      parcel = { weight: variant.weight, height: variant.height, width: variant.width, length: variant.length };
+
+    // Get default parcel size from primary shop
+    const { defaultParcelSize }  = Collections.Shops.findOne({
+      _id: shopId
+    });
+
+    // Set parcel
+    let parcel = {
+      weight: immediateAncestor ? immediateAncestor.weight : variant.weight,
```

Comment:
The logic here is wrong. If the immediate ancestor exists you use that weight even though that weight might be 0. Probably more correct (but still wrong) would be
```
weight: immediateAncestor.weight ? immediateAncestor.weight : variant.weight
```
but even that would be wrong because if the `variant.weight` exists we want to use that (that's why in my code it was applied last) so the correct logic would be

```
    let parcel = {
      weight: variant.weight ? variant.weight : immediateAncestor.weight,
      height: variant.height ? variant.height : immediateAncestor.height,
      width: variant.width ? variant.width : immediateAncestor.width,
      length: variant.length ? variant.length : immediateAncestor.length
    };
```

And a quick test of adding that to my cart shows that the correct get used instead of always using the default, but the default is still used when values are not defined on the variant.